cmd/clusterctl/cmd/delete.go

```diff
@@ -17,15 +17,82 @@ limitations under the License.
 package cmd
 
 import (
+	"github.com/pkg/errors"
 	"github.com/spf13/cobra"
 )
 
+type deleteOptions struct {
+	kubeconfig           string
+	forceDeleteNamespace bool
+	forceDeleteCRD       bool
+	deleteAll            bool
+}
+
+var dd = &deleteOptions{}
+
 var deleteCmd = &cobra.Command{
-	Use:   "delete",
-	Short: "Delete a cluster API resource",
-	Long:  `Delete a cluster API resource with one command`,
+	Use:   "delete [providers]",
+	Short: "Deletes one or more providers from the management cluster",
+	Long: LongDesc(`
+		Deletes one or more providers from the management cluster.`),
+
+	Example: Examples(`
+		# Deletes the AWS provider
+		# Please note that this imply the deletion of everything all the provider components except the 
+		# hosting namespace and the CRDs.
+		clusterctl delete aws
+ 
+		# Deletes the instance of the AWS provider hosted in the "foo" namespace
+		# Please note that, in case there are more than one instances of the AWS provider installed in the cluster,
+		# the provider components shared across providers (e.g. ClusterRoles), are not deleted in order to preserve
+		# the functioning of the remaining instances.
+		clusterctl delete foo/aws
+
+		# Deletes all the providers
+		# Important! As a consequence of this operation all the corresponding resources on target clouds
```

Comment:
"As a consequence of this operation, all the corresponding resources managed by Cluster API Providers are orphaned"

Reply:
done; BTW, the note about additional expenses was suggested by @detiber and I agree we should preserve it